# Load libraries
```{r}
.libPaths("/usr/lib/R/site-library")
library("edgeR", quietly = TRUE, lib.loc="/usr/lib/R/site-library")
library("DESeq2", quietly = TRUE, lib.loc="/usr/lib/R/site-library")
library("DEsingle", quietly = TRUE, lib.loc="/usr/lib/R/site-library")
library("biomaRt",quietly = TRUE, lib.loc="/usr/lib/R/site-library")
library("goseq" ,quietly = TRUE, lib.loc="/usr/lib/R/site-library")
library("VennDiagram", quietly = TRUE, lib.loc="/usr/lib/R/site-library")
library("ggpubr", quietly = TRUE, lib.loc="/usr/lib/R/site-library")
library("Rtsne", quietly = TRUE,lib.loc="/usr/lib/R/site-library")
library("MEGENA",quietly = TRUE, lib.loc="/usr/lib/R/site-library")
library("DGCA", quietly = TRUE, lib.loc="/usr/lib/R/site-library")
library('dplyr', quietly = TRUE,lib.loc="/usr/lib/R/site-library")
library("ggplot2", quietly = TRUE, lib.loc="/usr/lib/R/site-library")
library("ggrepel", quietly = TRUE, lib.loc="/usr/lib/R/site-library")
library("gridExtra",quietly = TRUE , lib.loc="/usr/lib/R/site-library")
library("readr" , quietly = TRUE, lib.loc="/usr/lib/R/site-library")
library("ggfortify" , quietly = TRUE, lib.loc="/usr/lib/R/site-library")
library("cowplot" , quietly = TRUE, lib.loc="/usr/lib/R/site-library")
library("tidyverse", quietly = TRUE, lib.loc="/usr/lib/R/site-library")
library("WGCNA", quietly = TRUE, lib.loc="/usr/lib/R/site-library")
library("reshape2", quietly = TRUE, lib.loc="/usr/lib/R/site-library")
library("foreach", quietly = TRUE, lib.loc="/usr/lib/R/site-library")
library("doParallel", quietly = TRUE, lib.loc="/usr/lib/R/site-library")
library("bigmemory", quietly = TRUE, lib.loc="/usr/lib/R/site-library")
library("gtools", quietly = TRUE, lib.loc="/usr/lib/R/site-library")
library("data.table", quietly = TRUE, lib.loc="/usr/lib/R/site-library")
library("flashClust", quietly = TRUE, lib.loc="/usr/lib/R/site-library")
library("dendextend", quietly = TRUE, lib.loc="/usr/lib/R/site-library")
library('vegan', quietly = TRUE, lib.loc="/usr/lib/R/site-library")
library("ade4", quietly = TRUE, lib.loc="/usr/lib/R/site-library")
library("gridGraphics", quietly = TRUE, lib.loc="/usr/lib/R/site-library")
library("networkD3", quietly = TRUE, lib.loc="/usr/lib/R/site-library")
library("readxl", quietly = TRUE, lib.loc="/usr/lib/R/site-library")
```

# Load the annotation of the genomne, transcripts , gene length , biological processes 


```{r}

annotation.ensembl.symbol <-read.table(file = "/home/yassine/Downloads/between_tissue/annotation.ensembl.symbol.txt",
                                       header= TRUE, 
                                       sep= "\t")

gene.length<-read.table(file = "/home/yassine/Downloads/between_tissue/gene.length.txt",
                                       header= TRUE, 
                                       sep= "\t")

annotation.GO.biomart<-read.table (file ="/home/yassine/Downloads/between_tissue/annotation.GO.biomart.txt",
                                       header= TRUE, 
                                       sep= "\t")
```

# Load the normalized filtred count data of the pwbc and the endometrium tissues 


```{r}
asinh_transf_CTF_normalized_pwbc_6<-read.table (file =                        "/home/yassine/Downloads/between_tissue/asinh_transf_CTF_normalized_pwbc_6.txt",
                                       header= TRUE, 
                                       sep= "\t")

asinh_transf_CTF_normalized_pwbc_9<-read.table (file =                        "/home/yassine/Downloads/between_tissue/asinh_transf_CTF_normalized_pwbc_9.txt",
                                       header= TRUE, 
                                       sep= "\t")
asinh_transf_CTF_normalized_pwbc_19<-read.table (file =                        "/home/yassine/Downloads/between_tissue/asinh_transf_CTF_normalized_pwbc_19.txt",
                                       header= TRUE, 
                                       sep= "\t")
```


```{r}
asinh_transf_CTF_normalized_endometrium_6<-read.table (file =                        "/home/yassine/Downloads/between_tissue/asinh_transf_CTF_normalized_endometrium_6.txt",
                                       header= TRUE, 
                                       sep= "\t")

asinh_transf_CTF_normalized_endometrium_9<-read.table (file =                        "/home/yassine/Downloads/between_tissue/asinh_transf_CTF_normalized_endometrium_9.txt",
                                       header= TRUE, 
                                       sep= "\t")

asinh_transf_CTF_normalized_endometrium_19<-read.table (file =                        "/home/yassine/Downloads/between_tissue/asinh_transf_CTF_normalized_endometrium_19.txt",
                                       header= TRUE, 
                                       sep= "\t")
```


# Venn diagramd to see the overlapping genes between pwbc and the endometrium tissues for each time point 
```{r}
pwbc_endometrium_6_venn_diagram<-venn.diagram(
  list("endometrium_6_months"=rownames(asinh_transf_CTF_normalized_endometrium_6),
       "pwbc_6_months"=rownames(asinh_transf_CTF_normalized_pwbc_6)),
                                        cex=1,
                                        scaled= FALSE, 
                                        filename = NULL,
                                        output=TRUE,
                                        col="transparent",
                                        fill=c("red","blue"))

pwbc_endometrium_9_venn_diagram<-venn.diagram(
  list("endometrium_9_months"=rownames(asinh_transf_CTF_normalized_endometrium_9),
       "pwbc_9_months"=rownames(asinh_transf_CTF_normalized_pwbc_9)),
                                        cex=1,
                                        scaled= FALSE, 
                                        filename = NULL,
                                        output=TRUE,
                                        col="transparent",
                                        fill=c("red","blue"))

pwbc_endometrium_19_venn_diagram<-venn.diagram(
  list("endometrium_19_months"=rownames(asinh_transf_CTF_normalized_endometrium_19),
       "pwbc_19_months"=rownames(asinh_transf_CTF_normalized_pwbc_19)),
                                        cex=1,
                                        scaled= FALSE, 
                                        filename = NULL,
                                        output=TRUE,
                                        col="transparent",
                                        fill=c("red","blue"))
```

```{r}
ggdraw(pwbc_endometrium_6_venn_diagram)
```

```{r}
ggdraw(pwbc_endometrium_9_venn_diagram)

```

```{r}
ggdraw(pwbc_endometrium_19_venn_diagram)

```

```{r}
ggsave(filename="/home/yassine/Downloads/between_tissue/Figures/pwbc_endometrium_6_venn_diagram.png",
       plot = pwbc_endometrium_6_venn_diagram, 
       width = 10, 
       height = 10)

ggsave(filename="/home/yassine/Downloads/between_tissue/Figures/pwbc_endometrium_9_venn_diagram.png",
       plot = pwbc_endometrium_9_venn_diagram, 
       width = 10, 
       height = 10)

ggsave(filename="/home/yassine/Downloads/between_tissue/Figures/pwbc_endometrium_19_venn_diagram.png",
       plot = pwbc_endometrium_19_venn_diagram, 
       width = 10, 
       height = 10)
```

# Use multiple cores 

```{r}
# Number of CPU cores to use
num_cores <- 4
# Register parallel cluster
cl <- makeCluster(num_cores)
registerDoParallel(cl)
```

# Plot the null distribution for the correlations between the two tissues 

```{r}
sample_a<-data.frame()
random_a<-data.frame()
random_b<-data.frame(matrix(NA, nrow = 122239488, ncol = 1))
for (i in (1:5)){
randomization<-sample(1:11,replace=F)

random_a<-WGCNA::cor(t(asinh_transf_CTF_normalized_endometrium_6[,randomization]), 
                     t(asinh_transf_CTF_normalized_pwbc_6),
                     use = "pairwise.complete.obs",
                     method="pearson")
random_a<-reshape2::melt(random_a) 
random_b<-cbind(random_b,random_a$value)
sample_a<-rbind(sample_a,randomization)
}

random_b<-random_b[,-1]
summary(random_b)

plot_null_distribution_pwbc_endometrium_6 <-ggplot()+
geom_histogram(aes(x=random_b[,1]), data=random_b, binwidth=0.01, alpha=0.2)+
geom_histogram(aes(x=random_b[,2]), data=random_b, binwidth=0.01, alpha=0.2)+
geom_histogram(aes(x=random_b[,3]), data=random_b, binwidth=0.01, alpha=0.2)+
geom_histogram(aes(x=random_b[,4]), data=random_b, binwidth=0.01, alpha=0.2)+
geom_histogram(aes(x=random_b[,5]), data=random_b, binwidth=0.01, alpha=0.2)+
scale_y_continuous(name="Count")+
scale_x_continuous(name="Correlation")+
theme(
title = "pwbc vs endometrium 6 month null distribution",
panel.background = element_blank(),
panel.grid.major = element_blank(),
plot.background = element_blank(),
axis.line = element_line(color="black"),
axis.text = element_text(color="black", size=10),
axis.title.y=element_text(color="black", size=10,hjust=0),
axis.title.x = element_text(color="black", size=10)
)
```
```{r}
sample_a<-data.frame()
random_a<-data.frame()
random_b<-data.frame(matrix(NA, nrow = 136157743, ncol = 1))
for (i in (1:5)){
randomization<-sample(1:11,replace=F)

random_a<-WGCNA::cor(t(asinh_transf_CTF_normalized_endometrium_9[,randomization]), 
                     t(asinh_transf_CTF_normalized_pwbc_9),
                     use = "pairwise.complete.obs",
                     method="pearson")
random_a<-reshape2::melt(random_a) 
random_b<-cbind(random_b,random_a$value)
sample_a<-rbind(sample_a,randomization)
}

random_b<-random_b[,-1]
summary(random_b)

plot_null_distribution_pwbc_endometrium_9 <-ggplot()+
geom_histogram(aes(x=random_b[,1]), data=random_b, binwidth=0.01, alpha=0.2)+
geom_histogram(aes(x=random_b[,2]), data=random_b, binwidth=0.01, alpha=0.2)+
geom_histogram(aes(x=random_b[,3]), data=random_b, binwidth=0.01, alpha=0.2)+
geom_histogram(aes(x=random_b[,4]), data=random_b, binwidth=0.01, alpha=0.2)+
geom_histogram(aes(x=random_b[,5]), data=random_b, binwidth=0.01, alpha=0.2)+
scale_y_continuous(name="Count")+
scale_x_continuous(name="Correlation")+
theme(
title = "pwbc vs endometrium 9 month null distribution",
panel.background = element_blank(),
panel.grid.major = element_blank(),
plot.background = element_blank(),
axis.line = element_line(color="black"),
axis.text = element_text(color="black", size=10),
axis.title.y=element_text(color="black", size=10,hjust=0),
axis.title.x = element_text(color="black", size=10)
)
```

```{r}
sample_a<-data.frame()
random_a<-data.frame()
random_b<-data.frame(matrix(NA, nrow = 134061792, ncol = 1))
for (i in (1:5)){
randomization<-sample(1:11,replace=F)

random_a<-WGCNA::cor(t(asinh_transf_CTF_normalized_endometrium_19[,randomization]), 
                     t(asinh_transf_CTF_normalized_pwbc_19),
                     use = "pairwise.complete.obs",
                     method="pearson")
random_a<-reshape2::melt(random_a) 
random_b<-cbind(random_b,random_a$value)
sample_a<-rbind(sample_a,randomization)
}

random_b<-random_b[,-1]
summary(random_b)

plot_null_distribution_pwbc_endometrium_19 <-ggplot()+
geom_histogram(aes(x=random_b[,1]), data=random_b, binwidth=0.01, alpha=0.2)+
geom_histogram(aes(x=random_b[,2]), data=random_b, binwidth=0.01, alpha=0.2)+
geom_histogram(aes(x=random_b[,3]), data=random_b, binwidth=0.01, alpha=0.2)+
geom_histogram(aes(x=random_b[,4]), data=random_b, binwidth=0.01, alpha=0.2)+
geom_histogram(aes(x=random_b[,5]), data=random_b, binwidth=0.01, alpha=0.2)+
scale_y_continuous(name="Count")+
scale_x_continuous(name="Correlation")+
theme(
title = "pwbc vs endometrium 19 month null distribution",
panel.background = element_blank(),
panel.grid.major = element_blank(),
plot.background = element_blank(),
axis.line = element_line(color="black"),
axis.text = element_text(color="black", size=10),
axis.title.y=element_text(color="black", size=10,hjust=0),
axis.title.x = element_text(color="black", size=10)
)
```


# Computing the correlation between the transformed filtred read counts of the pwbc and endometrium tissues 
```{r}
cor_pwbc_endometrium_6 <- corAndPvalue(t(asinh_transf_CTF_normalized_pwbc_6),
                                             t(asinh_transf_CTF_normalized_endometrium_6),
                                             use = "pairwise.complete.obs",
                                             alternative="two.sided")

cor_pwbc_endometrium_6_value<-reshape2::melt(cor_pwbc_endometrium_6$cor)
cor_pwbc_endometrium_6_value<-as.data.table(cor_pwbc_endometrium_6_value)
cor_pwbc_endometrium_6_p_value<-reshape2::melt(cor_pwbc_endometrium_6$p)
cor_pwbc_endometrium_p_value<-as.data.table(cor_pwbc_endometrium_6_p_value)
cor_pwbc_endometrium_6<-cbind(cor_pwbc_endometrium_6_value,cor_pwbc_endometrium_6_p_value)

rm(cor_pwbc_endometrium_6_p_value,cor_pwbc_endometrium_6_value)

cor_pwbc_endometrium_6_<-cor_pwbc_endometrium_6[,c(1,2,3,6)]
colnames(cor_pwbc_endometrium_6_)<-c("gene_pwbc",
                                            "gene_endometrium",
                                            "correlation",
                                            "p_value")

cor_pwbc_endometrium_6_<-setDT(cor_pwbc_endometrium_6_[order(cor_pwbc_endometrium_6_$p_value),])
```

```{r}
dim(cor_pwbc_endometrium_6_[abs(cor_pwbc_endometrium_6_$correlation)>0.9])

```

```{r}
cor_pwbc_endometrium_9 <- corAndPvalue(t(asinh_transf_CTF_normalized_pwbc_9),
                                             t(asinh_transf_CTF_normalized_endometrium_9),
                                             use = "pairwise.complete.obs",
                                             alternative="two.sided")

cor_pwbc_endometrium_9_value<-reshape2::melt(cor_pwbc_endometrium_9$cor)
cor_pwbc_endometrium_9_value<-as.data.table(cor_pwbc_endometrium_9_value)
cor_pwbc_endometrium_9_p_value<-reshape2::melt(cor_pwbc_endometrium_9$p)
cor_pwbc_endometrium_p_value<-as.data.table(cor_pwbc_endometrium_9_p_value)
cor_pwbc_endometrium_9<-cbind(cor_pwbc_endometrium_9_value,cor_pwbc_endometrium_9_p_value)

rm(cor_pwbc_endometrium_9_p_value,cor_pwbc_endometrium_9_value)

cor_pwbc_endometrium_9_<-cor_pwbc_endometrium_9[,c(1,2,3,6)]
colnames(cor_pwbc_endometrium_9_)<-c("gene_pwbc",
                                            "gene_endometrium",
                                            "correlation",
                                            "p_value")

cor_pwbc_endometrium_9_<-setDT(cor_pwbc_endometrium_9_[order(cor_pwbc_endometrium_9_$p_value),])
```

```{r}
dim(cor_pwbc_endometrium_9_[abs(cor_pwbc_endometrium_9_$correlation)>0.9])
```
```{r}
cor_pwbc_endometrium_19 <- corAndPvalue(t(asinh_transf_CTF_normalized_pwbc_19),
                                             t(asinh_transf_CTF_normalized_endometrium_19),
                                             use = "pairwise.complete.obs",
                                             alternative="two.sided")

cor_pwbc_endometrium_19_value<-reshape2::melt(cor_pwbc_endometrium_19$cor)
cor_pwbc_endometrium_19_value<-as.data.table(cor_pwbc_endometrium_19_value)
cor_pwbc_endometrium_19_p_value<-reshape2::melt(cor_pwbc_endometrium_19$p)
cor_pwbc_endometrium_p_value<-as.data.table(cor_pwbc_endometrium_19_p_value)
cor_pwbc_endometrium_19<-cbind(cor_pwbc_endometrium_19_value,cor_pwbc_endometrium_19_p_value)

rm(cor_pwbc_endometrium_19_p_value,cor_pwbc_endometrium_19_value)

cor_pwbc_endometrium_19_<-cor_pwbc_endometrium_19[,c(1,2,3,6)]
colnames(cor_pwbc_endometrium_19_)<-c("gene_pwbc",
                                            "gene_endometrium",
                                            "correlation",
                                            "p_value")

cor_pwbc_endometrium_19_<-setDT(cor_pwbc_endometrium_19_[order(cor_pwbc_endometrium_19_$p_value),])
```

```{r}
dim(cor_pwbc_endometrium_19_[abs(cor_pwbc_endometrium_19_$correlation)>0.9])

```

```{r}
cor_pwbc_endometrium_6_0.9<-cor_pwbc_endometrium_6_[abs(cor_pwbc_endometrium_6_$correlation)>0.9]

cor_pwbc_endometrium_6_0.9_annotated<-setDT(cor_pwbc_endometrium_6_0.9)

cor_pwbc_endometrium_6_0.9_annotated$pwbc_gene_symbol <- annotation.ensembl.symbol$external_gene_name[match(
  cor_pwbc_endometrium_6_0.9_annotated$gene_pwbc,
  annotation.ensembl.symbol$ensembl_gene_id)]

cor_pwbc_endometrium_6_0.9_annotated$endometrium_gene_symbol <- annotation.ensembl.symbol$external_gene_name[match(
  cor_pwbc_endometrium_6_0.9_annotated$gene_endometrium,
  annotation.ensembl.symbol$ensembl_gene_id)]
```


```{r}
cor_pwbc_endometrium_9_0.9<-cor_pwbc_endometrium_9_[abs(cor_pwbc_endometrium_9_$correlation)>0.9]

cor_pwbc_endometrium_9_0.9_annotated<-setDT(cor_pwbc_endometrium_9_0.9)

cor_pwbc_endometrium_9_0.9_annotated$pwbc_gene_symbol <- annotation.ensembl.symbol$external_gene_name[match(
  cor_pwbc_endometrium_9_0.9_annotated$gene_pwbc,
  annotation.ensembl.symbol$ensembl_gene_id)]

cor_pwbc_endometrium_9_0.9_annotated$endometrium_gene_symbol <- annotation.ensembl.symbol$external_gene_name[match(
  cor_pwbc_endometrium_9_0.9_annotated$gene_endometrium,
  annotation.ensembl.symbol$ensembl_gene_id)]
```

```{r}
cor_pwbc_endometrium_19_0.9<-cor_pwbc_endometrium_19_[abs(cor_pwbc_endometrium_19_$correlation)>0.9]

cor_pwbc_endometrium_19_0.9_annotated<-setDT(cor_pwbc_endometrium_19_0.9)

cor_pwbc_endometrium_19_0.9_annotated$pwbc_gene_symbol <- annotation.ensembl.symbol$external_gene_name[match(
  cor_pwbc_endometrium_19_0.9_annotated$gene_pwbc,
  annotation.ensembl.symbol$ensembl_gene_id)]

cor_pwbc_endometrium_19_0.9_annotated$endometrium_gene_symbol <- annotation.ensembl.symbol$external_gene_name[match(
  cor_pwbc_endometrium_19_0.9_annotated$gene_endometrium,
  annotation.ensembl.symbol$ensembl_gene_id)]
```

```{r}
write.table(cor_pwbc_endometrium_6_0.9_annotated,
            file = "/home/yassine/Downloads/between_tissue/cor_pwbc_endometrium_6_0.9_annotated_2023_10_27.txt",
            quote = TRUE,
            sep = "\t",
            row.names = TRUE,
            col.names = TRUE) 

write.table(cor_pwbc_endometrium_9_0.9_annotated,
            file = "/home/yassine/Downloads/between_tissue/cor_pwbc_endometrium_9_0.9_annotated_2023_10_27.txt",
            quote = TRUE,
            sep = "\t",
            row.names = TRUE,
            col.names = TRUE) 

write.table(cor_pwbc_endometrium_19_0.9_annotated,
            file = "/home/yassine/Downloads/between_tissue/cor_pwbc_endometrium_19_0.9_annotated_2023_10_27.txt",
            quote = TRUE,
            sep = "\t",
            row.names = TRUE,
            col.names = TRUE) 
```

# Plot the histogram for all values of coefficient.
```{r}
cor_pwbc_endometrium_6_histogram<-ggplot()+
geom_histogram(aes(x=correlation), 
               data=cor_pwbc_endometrium_6_,
               binwidth=0.01)+
scale_y_continuous(name="Count")+
scale_x_continuous(name="Correlation")+
theme(
      panel.background = element_blank(),
      panel.grid.major = element_blank(),
      plot.background = element_blank(),
      axis.line = element_line(color="black"),
      axis.text = element_text(color="black", size=10),
      axis.title.y=element_text(color="black", size=10,hjust=0),
      axis.title.x = element_text(color="black", size=10)
)
```


```{r}
cor_pwbc_endometrium_9_histogram<-ggplot()+
geom_histogram(aes(x=correlation), 
               data=cor_pwbc_endometrium_9_,
               binwidth=0.01)+
scale_y_continuous(name="Count")+
scale_x_continuous(name="Correlation")+
theme(
      panel.background = element_blank(),
      panel.grid.major = element_blank(),
      plot.background = element_blank(),
      axis.line = element_line(color="black"),
      axis.text = element_text(color="black", size=10),
      axis.title.y=element_text(color="black", size=10,hjust=0),
      axis.title.x = element_text(color="black", size=10)
)
```


```{r}
cor_pwbc_endometrium_19_histogram<-ggplot()+
geom_histogram(aes(x=correlation), 
               data=cor_pwbc_endometrium_19_,
               binwidth=0.01)+
scale_y_continuous(name="Count")+
scale_x_continuous(name="Correlation")+
theme(
      panel.background = element_blank(),
      panel.grid.major = element_blank(),
      plot.background = element_blank(),
      axis.line = element_line(color="black"),
      axis.text = element_text(color="black", size=10),
      axis.title.y=element_text(color="black", size=10,hjust=0),
      axis.title.x = element_text(color="black", size=10)
)
```


# Obtain gene connectivity.

```{r}
pwbc_6_gene_connectivity<- cor_pwbc_endometrium_6_0.9_annotated %>%
  dplyr::count(pwbc_gene_symbol) %>% 
  dplyr::arrange(desc(n))

pwbc_6_gene_connectivity<-pwbc_6_gene_connectivity[pwbc_6_gene_connectivity$n > 4,]
pwbc_6_gene_connectivity<-pwbc_6_gene_connectivity[!(pwbc_6_gene_connectivity$pwbc_gene_symbol ==""),]

pwbc_6_gene_connectivity$pwbc_gene_symbol<-factor(
  pwbc_6_gene_connectivity$pwbc_gene_symbol,
  levels=c(pwbc_6_gene_connectivity$pwbc_gene_symbol))

gene_connectivity_pwbc_6<-ggplot()+
geom_col( aes(x=pwbc_gene_symbol, y=n), 
          data=pwbc_6_gene_connectivity)+
  scale_x_discrete(name=NULL)+
  scale_y_continuous(name="Frequency")+
  #scale_y_break(c(100, 600 ) )+
  theme_classic()+
  theme(
    axis.text.x = element_text(angle=90, face="italic", size=7, vjust=0.5,hjust = 0),
    axis.text.y = element_text(color="black", size=10),
    axis.title.y=element_text(color="black", size=10,hjust=0),
    axis.title.x=element_blank()
  )
```

```{r}
gene_connectivity_pwbc_6
```
```{r}
pwbc_9_gene_connectivity<- cor_pwbc_endometrium_9_0.9_annotated %>%
  dplyr::count(pwbc_gene_symbol) %>% 
  dplyr::arrange(desc(n))
```


```{r}
pwbc_9_gene_connectivity<-pwbc_9_gene_connectivity[pwbc_9_gene_connectivity$n > 3,]
pwbc_9_gene_connectivity<-pwbc_9_gene_connectivity[!(pwbc_9_gene_connectivity$pwbc_gene_symbol ==""),]

pwbc_9_gene_connectivity$pwbc_gene_symbol<-factor(
  pwbc_9_gene_connectivity$pwbc_gene_symbol,
  levels=c(pwbc_9_gene_connectivity$pwbc_gene_symbol))

gene_connectivity_pwbc_9<-ggplot()+
geom_col( aes(x=pwbc_gene_symbol, y=n), 
          data=pwbc_9_gene_connectivity)+
  scale_x_discrete(name=NULL)+
  scale_y_continuous(name="Frequency")+
  #scale_y_break(c(100, 600 ) )+
  theme_classic()+
  theme(
    axis.text.x = element_text(angle=90, face="italic", size=7, vjust=0.5,hjust = 0),
    axis.text.y = element_text(color="black", size=10),
    axis.title.y=element_text(color="black", size=10,hjust=0),
    axis.title.x=element_blank()
  )
```


```{r}
pwbc_19_gene_connectivity<- cor_pwbc_endometrium_19_0.9_annotated %>%
  dplyr::count(pwbc_gene_symbol) %>% 
  dplyr::arrange(desc(n))

pwbc_19_gene_connectivity<-pwbc_19_gene_connectivity[pwbc_19_gene_connectivity$n > 4,]
pwbc_19_gene_connectivity<-pwbc_19_gene_connectivity[!(pwbc_19_gene_connectivity$pwbc_gene_symbol ==""),]

pwbc_19_gene_connectivity$pwbc_gene_symbol<-factor(
  pwbc_19_gene_connectivity$pwbc_gene_symbol,
  levels=c(pwbc_19_gene_connectivity$pwbc_gene_symbol))

gene_connectivity_pwbc_19<-ggplot()+
geom_col( aes(x=pwbc_gene_symbol, y=n), 
          data=pwbc_19_gene_connectivity)+
  scale_x_discrete(name=NULL)+
  scale_y_continuous(name="Frequency")+
  #scale_y_break(c(100, 600 ) )+
  theme_classic()+
  theme(
    axis.text.x = element_text(angle=90, face="italic", size=7, vjust=0.5,hjust = 0),
    axis.text.y = element_text(color="black", size=10),
    axis.title.y=element_text(color="black", size=10,hjust=0),
    axis.title.x=element_blank()
  )
```


```{r}
endometrium_6_gene_connectivity<- cor_pwbc_endometrium_6_0.9_annotated %>%
  dplyr::count(endometrium_gene_symbol) %>% 
  dplyr::arrange(desc(n))

endometrium_6_gene_connectivity<-endometrium_6_gene_connectivity[endometrium_6_gene_connectivity$n > 4,]
endometrium_6_gene_connectivity<-endometrium_6_gene_connectivity[!(endometrium_6_gene_connectivity$endometrium_gene_symbol ==""),]

endometrium_6_gene_connectivity$endometrium_gene_symbol<-factor(endometrium_6_gene_connectivity$endometrium_gene_symbol,
                                                levels=c(endometrium_6_gene_connectivity$endometrium_gene_symbol))


gene_connectivity_endometrium_6<-ggplot()+
geom_col( aes(x=endometrium_gene_symbol, y=n), 
          data=endometrium_6_gene_connectivity)+
  scale_x_discrete(name=NULL)+
  scale_y_continuous(name="Frequency")+
  #scale_y_break(c(100, 600 ) )+
  theme_classic()+
  theme(
    axis.text.x = element_text(angle=90, face="italic", size=7, vjust=0.5,hjust = 0),
    axis.text.y = element_text(color="black", size=10),
    axis.title.y=element_text(color="black", size=10,hjust=0),
    axis.title.x=element_blank()
  )

```

```{r}
endometrium_9_gene_connectivity<- cor_pwbc_endometrium_9_0.9_annotated %>%
  dplyr::count(endometrium_gene_symbol) %>% 
  dplyr::arrange(desc(n))

endometrium_9_gene_connectivity<-endometrium_9_gene_connectivity[endometrium_9_gene_connectivity$n > 4,]
endometrium_9_gene_connectivity<-endometrium_9_gene_connectivity[!(endometrium_9_gene_connectivity$endometrium_gene_symbol ==""),]

endometrium_9_gene_connectivity$endometrium_gene_symbol<-factor(endometrium_9_gene_connectivity$endometrium_gene_symbol,
                                                levels=c(endometrium_9_gene_connectivity$endometrium_gene_symbol))


gene_connectivity_endometrium_9<-ggplot()+
geom_col( aes(x=endometrium_gene_symbol, y=n), 
          data=endometrium_9_gene_connectivity)+
  scale_x_discrete(name=NULL)+
  scale_y_continuous(name="Frequency")+
  #scale_y_break(c(100, 600 ) )+
  theme_classic()+
  theme(
    axis.text.x = element_text(angle=90, face="italic", size=7, vjust=0.5,hjust = 0),
    axis.text.y = element_text(color="black", size=10),
    axis.title.y=element_text(color="black", size=10,hjust=0),
    axis.title.x=element_blank()
  )
```

```{r}
endometrium_19_gene_connectivity<- cor_pwbc_endometrium_19_0.9_annotated %>%
  dplyr::count(endometrium_gene_symbol) %>% 
  dplyr::arrange(desc(n))

endometrium_19_gene_connectivity<-endometrium_19_gene_connectivity[endometrium_19_gene_connectivity$n > 4,]
endometrium_19_gene_connectivity<-endometrium_19_gene_connectivity[!(endometrium_19_gene_connectivity$endometrium_gene_symbol ==""),]

endometrium_19_gene_connectivity$endometrium_gene_symbol<-factor(endometrium_19_gene_connectivity$endometrium_gene_symbol,
                                                levels=c(endometrium_19_gene_connectivity$endometrium_gene_symbol))


gene_connectivity_endometrium_19<-ggplot()+
geom_col( aes(x=endometrium_gene_symbol, y=n), 
          data=endometrium_19_gene_connectivity)+
  scale_x_discrete(name=NULL)+
  scale_y_continuous(name="Frequency")+
  #scale_y_break(c(100, 600 ) )+
  theme_classic()+
  theme(
    axis.text.x = element_text(angle=90, face="italic", size=7, vjust=0.5,hjust = 0),
    axis.text.y = element_text(color="black", size=10),
    axis.title.y=element_text(color="black", size=10,hjust=0),
    axis.title.x=element_blank()
  )
```



# Plot examples of co-expressed genes between pwbc and endometrium tissues
```{r}
cor_pwbc_endometrium_6_0.9_filtered <- cor_pwbc_endometrium_6_0.9

cor_pwbc_endometrium_6_0.9_filtered$pwbc_gene_symbol <- annotation.ensembl.symbol$external_gene_name[match(
  cor_pwbc_endometrium_6_0.9_filtered$gene_pwbc,
  annotation.ensembl.symbol$ensembl_gene_id)]

cor_pwbc_endometrium_6_0.9_filtered$endometrium_gene_symbol <-
  annotation.ensembl.symbol$external_gene_name[match(
    cor_pwbc_endometrium_6_0.9_filtered$gene_endometrium,
    annotation.ensembl.symbol$ensembl_gene_id)]

cor_pwbc_endometrium_6_0.9_filtered$angle<-NA
cor_pwbc_endometrium_6_0.9_filtered$r.squared<-NA
cor_pwbc_endometrium_6_0.9_filtered$rmse<-NA
cor_pwbc_endometrium_6_0.9_filtered$fstatistic<-NA

for (i in seq(dim(cor_pwbc_endometrium_6_0.9_filtered)[1]))
{
  gene_pwbc<-as.character(cor_pwbc_endometrium_6_0.9_filtered$gene_pwbc)[i]
  gene_endometrium<-as.character(cor_pwbc_endometrium_6_0.9_filtered$gene_endometrium)[i]
  
  gene_symbol_pwbc<-as.character(cor_pwbc_endometrium_6_0.9_filtered$pwbc_gene_symbol)[i]
  gene_symbol_endometrium<-as.character(cor_pwbc_endometrium_6_0.9_filtered$endometrium_gene_symbol)[i]
  
  expression_pwbc<-data.frame(expression_pwbc=t(pwbc_6_tpm_filtered[rownames(pwbc_6_tpm_filtered)==gene_pwbc,])[,1])
  expression_endometrium<-data.frame(expression_endometrium=t(endometrium_6_tpm_filtered[rownames(endometrium_6_tpm_filtered)==gene_endometrium,])[,1])
  
  model_pwbc_endo <-lm(expression_pwbc$expression_pwbc ~ expression_endometrium$expression_endometrium)
  cor_pwbc_endometrium_6_0.9_filtered$angle[i]<-unname(atan(coef(model_pwbc_endo)[2]) * (180 / pi))
  cor_pwbc_endometrium_6_0.9_filtered$r.squared[i]<-summary(model_pwbc_endo)$adj.r.squared
  cor_pwbc_endometrium_6_0.9_filtered$rmse[i]<-sqrt(mean(model_pwbc_endo$residuals^2))
  cor_pwbc_endometrium_6_0.9_filtered$fstatistic[i]<-unname(summary(model_pwbc_endo)$fstatistic[1])
}

cor_pwbc_endometrium_6_0.9_filtered<-cor_pwbc_endometrium_6_0.9_filtered[cor_pwbc_endometrium_6_0.9_filtered$angle > 40 & cor_pwbc_endometrium_6_0.9_filtered$angle < 60,]

cor_pwbc_endometrium_6_0.9_filtered<-cor_pwbc_endometrium_6_0.9_filtered[order(cor_pwbc_endometrium_6_0.9_filtered$rmse , decreasing = FALSE),]

cor_pwbc_endometrium_6_0.9_filtered<-cor_pwbc_endometrium_6_0.9_filtered[cor_pwbc_endometrium_6_0.9_filtered$r.squared>0.3,]

```

```{r}
cor_pwbc_endometrium_6_0.9_filtered
```


```{r}
dataframegraph<-data.frame()

for (i in c(1:10)){

  gene_pwbc<-as.character(cor_pwbc_endometrium_6_0.9_filtered$gene_pwbc)[i]
  gene_endometrium<-as.character(cor_pwbc_endometrium_6_0.9_filtered$gene_endometrium)[i]
  
  gene_symbol_pwbc<-as.character(cor_pwbc_endometrium_6_0.9_filtered$pwbc_gene_symbol)[i]
  gene_symbol_endometrium<-as.character(cor_pwbc_endometrium_6_0.9_filtered$endometrium_gene_symbol)[i]
  
  expression_pwbc<-data.frame(expression_pwbc=t(pwbc_6_tpm_filtered[rownames(pwbc_6_tpm_filtered)==gene_pwbc,])[,1])
  
  expression_endometrium<-data.frame( expression_endometrium=t(endometrium_6_tpm_filtered[rownames(endometrium_6_tpm_filtered)==gene_endometrium,])[,1])
  
  graph<-i
  
  dataframegraph<-rbind(data.frame(gene_pwbc,
                                   gene_endometrium,
                                   gene_symbol_pwbc,
                                   gene_symbol_endometrium,
                                   expression_pwbc,
                                   expression_endometrium,
                                   graph),
                        dataframegraph)

}
plots<-list()
for (i in c(1:10)){
  plots[[i]]<- ggplot(dataframegraph[dataframegraph$graph==i,],
                      aes(x=expression_pwbc,
                          y= expression_endometrium))+
  geom_point()+
  geom_smooth(method='lm', formula= y~x)+
  scale_x_continuous(name=dataframegraph[dataframegraph$graph==i,]$gene_symbol_pwbc[1])+
  scale_y_continuous(name=dataframegraph[dataframegraph$graph==i,]$gene_symbol_endometrium[1])+
  theme_classic(base_size=12)+
  theme(
  axis.title=element_text(face="italic", size=10),
  axis.text=element_text(size=10, color="black"),
  )

}

x.grob <- textGrob("pwbc",
                   gp=gpar(fontface="plain", col="black", fontsize=10))
y.grob <- textGrob("endometrium",
                   gp=gpar(fontface="plain", col="black", fontsize=10),
                   rot=90)
```


```{r}
plot_regr_6<-plot_grid( grid.arrange(arrangeGrob(plot_grid(plotlist=plots, nrow=2),
                                               left = y.grob,
                                               bottom = x.grob)))
```



```{r}
cor_pwbc_endometrium_9_0.9_filtered <- cor_pwbc_endometrium_9_0.9

cor_pwbc_endometrium_9_0.9_filtered$pwbc_gene_symbol <- annotation.ensembl.symbol$external_gene_name[match(
  cor_pwbc_endometrium_9_0.9_filtered$gene_pwbc,
  annotation.ensembl.symbol$ensembl_gene_id)]

cor_pwbc_endometrium_9_0.9_filtered$endometrium_gene_symbol <-
  annotation.ensembl.symbol$external_gene_name[match(
    cor_pwbc_endometrium_9_0.9_filtered$gene_endometrium,
    annotation.ensembl.symbol$ensembl_gene_id)]

cor_pwbc_endometrium_9_0.9_filtered$angle<-NA
cor_pwbc_endometrium_9_0.9_filtered$r.squared<-NA
cor_pwbc_endometrium_9_0.9_filtered$rmse<-NA
cor_pwbc_endometrium_9_0.9_filtered$fstatistic<-NA

for (i in seq(dim(cor_pwbc_endometrium_9_0.9_filtered)[1]))
{
  gene_pwbc<-as.character(cor_pwbc_endometrium_9_0.9_filtered$gene_pwbc)[i]
  gene_endometrium<-as.character(cor_pwbc_endometrium_9_0.9_filtered$gene_endometrium)[i]
  
  gene_symbol_pwbc<-as.character(cor_pwbc_endometrium_9_0.9_filtered$pwbc_gene_symbol)[i]
  gene_symbol_endometrium<-as.character(cor_pwbc_endometrium_9_0.9_filtered$endometrium_gene_symbol)[i]
  
  expression_pwbc<-data.frame(expression_pwbc=t(pwbc_9_tpm_filtered[rownames(pwbc_9_tpm_filtered)==gene_pwbc,])[,1])
  expression_endometrium<-data.frame(expression_endometrium=t(endometrium_9_tpm_filtered[rownames(endometrium_9_tpm_filtered)==gene_endometrium,])[,1])
  
  model_pwbc_endo <-lm(expression_pwbc$expression_pwbc ~ expression_endometrium$expression_endometrium)
  cor_pwbc_endometrium_9_0.9_filtered$angle[i]<-unname(atan(coef(model_pwbc_endo)[2]) * (180 / pi))
  cor_pwbc_endometrium_9_0.9_filtered$r.squared[i]<-summary(model_pwbc_endo)$adj.r.squared
  cor_pwbc_endometrium_9_0.9_filtered$rmse[i]<-sqrt(mean(model_pwbc_endo$residuals^2))
  cor_pwbc_endometrium_9_0.9_filtered$fstatistic[i]<-unname(summary(model_pwbc_endo)$fstatistic[1])
}

cor_pwbc_endometrium_9_0.9_filtered<-cor_pwbc_endometrium_9_0.9_filtered[cor_pwbc_endometrium_9_0.9_filtered$angle > 0 & cor_pwbc_endometrium_9_0.9_filtered$angle < 90,]

cor_pwbc_endometrium_9_0.9_filtered<-cor_pwbc_endometrium_9_0.9_filtered[order(cor_pwbc_endometrium_9_0.9_filtered$rmse , decreasing = FALSE),]

cor_pwbc_endometrium_9_0.9_filtered<-cor_pwbc_endometrium_9_0.9_filtered[cor_pwbc_endometrium_9_0.9_filtered$r.squared>0.3,]

```

```{r}
cor_pwbc_endometrium_9_0.9_filtered
```


```{r}
dataframegraph<-data.frame()

for (i in c(1:9)){

  gene_pwbc<-as.character(cor_pwbc_endometrium_9_0.9_filtered$gene_pwbc)[i]
  gene_endometrium<-as.character(cor_pwbc_endometrium_9_0.9_filtered$gene_endometrium)[i]
  
  gene_symbol_pwbc<-as.character(cor_pwbc_endometrium_9_0.9_filtered$pwbc_gene_symbol)[i]
  gene_symbol_endometrium<-as.character(cor_pwbc_endometrium_9_0.9_filtered$endometrium_gene_symbol)[i]
  
  expression_pwbc<-data.frame(expression_pwbc=t(pwbc_9_tpm_filtered[rownames(pwbc_9_tpm_filtered)==gene_pwbc,])[,1])
  
  expression_endometrium<-data.frame( expression_endometrium=t(endometrium_9_tpm_filtered[rownames(endometrium_9_tpm_filtered)==gene_endometrium,])[,1])
  
  graph<-i
  
  dataframegraph<-rbind(data.frame(gene_pwbc,
                                   gene_endometrium,
                                   gene_symbol_pwbc,
                                   gene_symbol_endometrium,
                                   expression_pwbc,
                                   expression_endometrium,
                                   graph),
                        dataframegraph)

}
plots<-list()
for (i in c(1:9)){
  plots[[i]]<- ggplot(dataframegraph[dataframegraph$graph==i,],
                      aes(x=expression_pwbc,
                          y= expression_endometrium))+
  geom_point()+
  geom_smooth(method='lm', formula= y~x)+
  scale_x_continuous(name=dataframegraph[dataframegraph$graph==i,]$gene_symbol_pwbc[1])+
  scale_y_continuous(name=dataframegraph[dataframegraph$graph==i,]$gene_symbol_endometrium[1])+
  theme_classic(base_size=12)+
  theme(
  axis.title=element_text(face="italic", size=10),
  axis.text=element_text(size=10, color="black"),
  )

}

x.grob <- textGrob("pwbc",
                   gp=gpar(fontface="plain", col="black", fontsize=10))
y.grob <- textGrob("endometrium",
                   gp=gpar(fontface="plain", col="black", fontsize=10),
                   rot=90)
```


```{r}
plot_regr_9<-plot_grid( grid.arrange(arrangeGrob(plot_grid(plotlist=plots, nrow=3),
                                               left = y.grob,
                                               bottom = x.grob)))
```

```{r}
cor_pwbc_endometrium_19_0.9_filtered <- cor_pwbc_endometrium_19_0.9

cor_pwbc_endometrium_19_0.9_filtered$pwbc_gene_symbol <- annotation.ensembl.symbol$external_gene_name[match(
  cor_pwbc_endometrium_19_0.9_filtered$gene_pwbc,
  annotation.ensembl.symbol$ensembl_gene_id)]

cor_pwbc_endometrium_19_0.9_filtered$endometrium_gene_symbol <-
  annotation.ensembl.symbol$external_gene_name[match(
    cor_pwbc_endometrium_19_0.9_filtered$gene_endometrium,
    annotation.ensembl.symbol$ensembl_gene_id)]

cor_pwbc_endometrium_19_0.9_filtered$angle<-NA
cor_pwbc_endometrium_19_0.9_filtered$r.squared<-NA
cor_pwbc_endometrium_19_0.9_filtered$rmse<-NA
cor_pwbc_endometrium_19_0.9_filtered$fstatistic<-NA

for (i in seq(dim(cor_pwbc_endometrium_19_0.9_filtered)[1]))
{
  gene_pwbc<-as.character(cor_pwbc_endometrium_19_0.9_filtered$gene_pwbc)[i]
  gene_endometrium<-as.character(cor_pwbc_endometrium_19_0.9_filtered$gene_endometrium)[i]
  
  gene_symbol_pwbc<-as.character(cor_pwbc_endometrium_19_0.9_filtered$pwbc_gene_symbol)[i]
  gene_symbol_endometrium<-as.character(cor_pwbc_endometrium_19_0.9_filtered$endometrium_gene_symbol)[i]
  
  expression_pwbc<-data.frame(expression_pwbc=t(pwbc_19_tpm_filtered[rownames(pwbc_19_tpm_filtered)==gene_pwbc,])[,1])
  expression_endometrium<-data.frame(expression_endometrium=t(endometrium_19_tpm_filtered[rownames(endometrium_19_tpm_filtered)==gene_endometrium,])[,1])
  
  model_pwbc_endo <-lm(expression_pwbc$expression_pwbc ~ expression_endometrium$expression_endometrium)
  cor_pwbc_endometrium_19_0.9_filtered$angle[i]<-unname(atan(coef(model_pwbc_endo)[2]) * (180 / pi))
  cor_pwbc_endometrium_19_0.9_filtered$r.squared[i]<-summary(model_pwbc_endo)$adj.r.squared
  cor_pwbc_endometrium_19_0.9_filtered$rmse[i]<-sqrt(mean(model_pwbc_endo$residuals^2))
  cor_pwbc_endometrium_19_0.9_filtered$fstatistic[i]<-unname(summary(model_pwbc_endo)$fstatistic[1])
}

cor_pwbc_endometrium_19_0.9_filtered<-cor_pwbc_endometrium_19_0.9_filtered[cor_pwbc_endometrium_19_0.9_filtered$angle > 40 & cor_pwbc_endometrium_19_0.9_filtered$angle < 60,]

cor_pwbc_endometrium_19_0.9_filtered<-cor_pwbc_endometrium_19_0.9_filtered[order(cor_pwbc_endometrium_19_0.9_filtered$rmse , decreasing = FALSE),]

cor_pwbc_endometrium_19_0.9_filtered<-cor_pwbc_endometrium_19_0.9_filtered[cor_pwbc_endometrium_19_0.9_filtered$r.squared>0.3,]

```



```{r}
dataframegraph<-data.frame()

for (i in c(1:10)){

  gene_pwbc<-as.character(cor_pwbc_endometrium_19_0.9_filtered$gene_pwbc)[i]
  gene_endometrium<-as.character(cor_pwbc_endometrium_19_0.9_filtered$gene_endometrium)[i]
  
  gene_symbol_pwbc<-as.character(cor_pwbc_endometrium_19_0.9_filtered$pwbc_gene_symbol)[i]
  gene_symbol_endometrium<-as.character(cor_pwbc_endometrium_19_0.9_filtered$endometrium_gene_symbol)[i]
  
  expression_pwbc<-data.frame(expression_pwbc=t(pwbc_19_tpm_filtered[rownames(pwbc_19_tpm_filtered)==gene_pwbc,])[,1])
  
  expression_endometrium<-data.frame( expression_endometrium=t(endometrium_19_tpm_filtered[rownames(endometrium_19_tpm_filtered)==gene_endometrium,])[,1])
  
  graph<-i
  
  dataframegraph<-rbind(data.frame(gene_pwbc,
                                   gene_endometrium,
                                   gene_symbol_pwbc,
                                   gene_symbol_endometrium,
                                   expression_pwbc,
                                   expression_endometrium,
                                   graph),
                        dataframegraph)

}
plots<-list()
for (i in c(1:10)){
  plots[[i]]<- ggplot(dataframegraph[dataframegraph$graph==i,],
                      aes(x=expression_pwbc,
                          y= expression_endometrium))+
  geom_point()+
  geom_smooth(method='lm', formula= y~x)+
  scale_x_continuous(name=dataframegraph[dataframegraph$graph==i,]$gene_symbol_pwbc[1])+
  scale_y_continuous(name=dataframegraph[dataframegraph$graph==i,]$gene_symbol_endometrium[1])+
  theme_classic(base_size=12)+
  theme(
  axis.title=element_text(face="italic", size=10),
  axis.text=element_text(size=10, color="black"),
  )

}

x.grob <- textGrob("pwbc",
                   gp=gpar(fontface="plain", col="black", fontsize=10))
y.grob <- textGrob("endometrium",
                   gp=gpar(fontface="plain", col="black", fontsize=10),
                   rot=90)
```


```{r}
plot_regr_19<-plot_grid( grid.arrange(arrangeGrob(plot_grid(plotlist=plots, nrow=2),
                                               left = y.grob,
                                               bottom = x.grob)))
```

# Test endometrium genes (significantly co-expressed with pwbc) for enrichment of biological processes.

```{r}
all_genes_endometrium_6 <-data.frame( gene=rownames(endometrium_6_count_filtered),
                                    stringsAsFactors=FALSE )

rownames(all_genes_endometrium_6)<-all_genes_endometrium_6$gene
N_expressed_genes<-length(all_genes_endometrium_6$gene)

gene.length_endometrium_6 <-gene.length[gene.length$ensembl_gene_id %in% all_genes_endometrium_6$gene,]

annotation.genelength.biomart_vector_endometrium_6 <-gene.length_endometrium_6$transcript_length
names(annotation.genelength.biomart_vector_endometrium_6)<-gene.length_endometrium_6$ensembl_gene_id

annotation.GO.BP.biomart<-annotation.GO.biomart[annotation.GO.biomart$namespace_1003=="biological_process", c(1,3)]
annotation.GO.BP.biomart<-annotation.GO.BP.biomart[annotation.GO.BP.biomart$ensembl_gene_id %in% rownames(all_genes_endometrium_6),]
annotation.GO.MF.biomart<-annotation.GO.biomart[annotation.GO.biomart$namespace_1003=="molecular_function", c(1,3)]
annotation.GO.MF.biomart<-annotation.GO.MF.biomart[annotation.GO.MF.biomart$ensembl_gene_id %in% rownames(all_genes_endometrium_6),]

```

```{r}
test.genes_endometrium_6 <-data.frame(
  a=unique(cor_pwbc_endometrium_6_[abs(cor_pwbc_endometrium_6_$correlation)>0.9]$gene_endometrium), 
  stringsAsFactors=FALSE)

all_genes_endometrium_6_numeric<-as.integer(all_genes_endometrium_6$gene %in% test.genes_endometrium_6$a)
names(all_genes_endometrium_6_numeric)<-all_genes_endometrium_6$gene
N_sig_genes<-length(test.genes_endometrium_6$a)

```

```{r}
set.seed(88972)

pwf_endometrium_6 <- nullp(all_genes_endometrium_6_numeric,
                            bias.data=annotation.genelength.biomart_vector_endometrium_6,
                            plot.fit=FALSE )
```


```{r}
GO_BP_Cats_endometrium_corr_pwbc_6 <- goseq(pwf_endometrium_6,
                                                  gene2cat=annotation.GO.BP.biomart,
                                                  method ="Sampling",
                                                  repcnt = 5000,
                                                  use_genes_without_cat=FALSE)
```

```{r}
write.table(GO_BP_Cats_endometrium_corr_pwbc_6,
            file = "/home/yassine/Downloads/between_tissue/GO_BP_Cats_endometrium_corr_pwbc_6.txt",
            quote = TRUE,
            sep = "\t",
            row.names = TRUE,
            col.names = TRUE) 
```

```{r}

GO_BP_Cats_endometrium_corr_pwbc_6<-GO_BP_Cats_endometrium_corr_pwbc_6[GO_BP_Cats_endometrium_corr_pwbc_6$numInCat>3,] 

GO_BP_Cats_endometrium_corr_pwbc_6$FWER<-p.adjust(
  GO_BP_Cats_endometrium_corr_pwbc_6$over_represented_pvalue,
  method ="holm")

GO_BP_Cats_endometrium_corr_pwbc_6<-GO_BP_Cats_endometrium_corr_pwbc_6[with(GO_BP_Cats_endometrium_corr_pwbc_6,order(FWER,over_represented_pvalue,-numInCat)), ]

GO_BP_Cats_endometrium_corr_pwbc_6$fold_enrichment<-(GO_BP_Cats_endometrium_corr_pwbc_6$numDEInCat/N_sig_genes)/(GO_BP_Cats_endometrium_corr_pwbc_6$numInCat/N_expressed_genes)
```



```{r}
annotation.GO.BP.biomart_testgenes<-annotation.GO.BP.biomart[annotation.GO.BP.biomart$ensembl_gene_id %in% test.genes_endometrium_6$a, ]

GO_BP_Cats_endometrium_corr_pwbc_6<-merge(GO_BP_Cats_endometrium_corr_pwbc_6,
                                                annotation.GO.BP.biomart_testgenes,
                                                by.x="category",
                                                by.y="go_id",
                                                all.x=FALSE,
                                                all.y=TRUE)

GO_BP_Cats_endometrium_corr_pwbc_6<-merge(GO_BP_Cats_endometrium_corr_pwbc_6,
                                                annotation.ensembl.symbol,
                                                by.x="ensembl_gene_id",
                                                by.y="ensembl_gene_id", 
                                                all=FALSE, 
                                                all.x=TRUE,
                                                all.y=FALSE)
```

```{r}
GO_BP_Cats_endometrium_corr_pwbc_6 <- arrange(GO_BP_Cats_endometrium_corr_pwbc_6, desc(fold_enrichment))

```

```{r}

GO_BP_Cats_endometrium_corr_pwbc_6$term<-factor(
  GO_BP_Cats_endometrium_corr_pwbc_6$term,
  levels=c(rev(unique(GO_BP_Cats_endometrium_corr_pwbc_6$term))))
```



# Test pwbc genes (significantly co-expressed with endometrium cells) for enrichment of biological processes.

```{r}
all_genes_pwbc_6 <-data.frame( gene=rownames(pwbc_6_count_filtered),
                                    stringsAsFactors=FALSE )

rownames(all_genes_pwbc_6) <- all_genes_pwbc_6$gene
N_expressed_genes <- length(all_genes_pwbc_6$gene)

gene.length_pwbc_6 <- gene.length[gene.length$ensembl_gene_id %in% all_genes_pwbc_6$gene,]

annotation.genelength.biomart_vector_pwbc_6 <- gene.length_pwbc_6$transcript_length
names(annotation.genelength.biomart_vector_pwbc_6) <- gene.length_pwbc_6$ensembl_gene_id

annotation.GO.BP.biomart<-annotation.GO.biomart[annotation.GO.biomart$namespace_1003=="biological_process", c(1,3)]

annotation.GO.BP.biomart<-annotation.GO.BP.biomart[annotation.GO.BP.biomart$ensembl_gene_id %in% rownames(all_genes_pwbc_6),]

annotation.GO.MF.biomart<-annotation.GO.biomart[annotation.GO.biomart$namespace_1003=="molecular_function", c(1,3)]

annotation.GO.MF.biomart<-annotation.GO.MF.biomart[annotation.GO.MF.biomart$ensembl_gene_id %in% rownames(all_genes_pwbc_6),]
```


```{r}
test.genes_pwbc_6 <-data.frame(
  a=unique(cor_pwbc_endometrium_6_[abs(cor_pwbc_endometrium_6_$correlation)>0.9]$gene_pwbc),
  stringsAsFactors=FALSE)

all_genes_pwbc_6_numeric<-as.integer(all_genes_pwbc_6$gene %in% test.genes_pwbc_6$a)
names(all_genes_pwbc_6_numeric)<-all_genes_pwbc_6$gene

N_sig_genes<-length(test.genes_pwbc_6$a)
```



```{r}
set.seed(87175)

pwf_pwbc_6 <-nullp(all_genes_pwbc_6_numeric,
                        bias.data=annotation.genelength.biomart_vector_pwbc_6,
                        plot.fit=FALSE )
```


```{r}
GO_BP_Cats_pwbc_corr_endometrium_6 <-goseq(pwf_pwbc_6,
                                                 gene2cat=annotation.GO.BP.biomart,
                                                 method ="Sampling",
                                                 repcnt = 5000,
                                                 use_genes_without_cat=FALSE)
```
```{r}
GO_BP_Cats_pwbc_corr_endometrium_6<-GO_BP_Cats_pwbc_corr_endometrium_6[GO_BP_Cats_pwbc_corr_endometrium_6$numInCat>3,]

GO_BP_Cats_pwbc_corr_endometrium_6$FWER<-p.adjust(
  GO_BP_Cats_pwbc_corr_endometrium_6$over_represented_pvalue, 
  method ="holm")

GO_BP_Cats_pwbc_corr_endometrium_6<-GO_BP_Cats_pwbc_corr_endometrium_6[
  with(GO_BP_Cats_pwbc_corr_endometrium_6,
       order(FWER,over_represented_pvalue,-numInCat)), ]

GO_BP_Cats_pwbc_corr_endometrium_6$fold_enrichment<-(GO_BP_Cats_pwbc_corr_endometrium_6$numDEInCat/N_sig_genes)/(GO_BP_Cats_pwbc_corr_endometrium_6$numInCat/N_expressed_genes)
                                                                                                
annotation.GO.BP.biomart_testgenes<-annotation.GO.BP.biomart[annotation.GO.BP.biomart$ensembl_gene_id %in% test.genes_pwbc_6$a, ]
```


```{r}
GO_BP_Cats_pwbc_corr_endometrium_6<-merge(GO_BP_Cats_pwbc_corr_endometrium_6,
                                              annotation.GO.BP.biomart_testgenes,
                                              by.x="category",
                                              by.y="go_id",
                                              all.x=TRUE,
                                              all.y=FALSE)

GO_BP_Cats_pwbc_corr_endometrium_6<-merge(GO_BP_Cats_pwbc_corr_endometrium_6,
                                              annotation.ensembl.symbol,
                                              by.x="ensembl_gene_id",
                                              by.y="ensembl_gene_id",
                                              all=FALSE, 
                                              all.x=TRUE,
                                              all.y=FALSE)
```


```{r}
GO_BP_Cats_pwbc_corr_endometrium_6 <- arrange(GO_BP_Cats_pwbc_corr_endometrium_6, desc(fold_enrichment))

```


```{r}
GO_BP_Cats_pwbc_corr_endometrium_6$term<-factor(
  GO_BP_Cats_pwbc_corr_endometrium_6$term,
  levels=c(rev(unique(GO_BP_Cats_pwbc_corr_endometrium_6$term))))
```


```{r}
GO_pwbc_6_plot <- ggplot(data = GO_BP_Cats_pwbc_corr_endometrium_6[1:10, ],
                             aes(x = fold_enrichment,
                                y = term)) +
  geom_point() +
  coord_cartesian() +
  scale_colour_gradient(name = "Fold enrichment",
                        low = "#B8B8B8",
                        high = "#080808",
                        na.value = NA) +
  scale_x_continuous(name = "Fold enrichment") +
  scale_y_discrete(name = "Biological process") +
  theme_classic() +
  ggtitle("pwbc 6 months - Biological processes") +
  theme(
    legend.direction = "horizontal",
    legend.position = c(0.9, 1),
    legend.key.size = unit(0.5, "mm"),
    legend.text = element_text(size = 5, margin = margin(t = 0, r = 0, b = 0, l = 0, unit = "cm")),
    legend.title = element_text(size = 5),
    legend.background = element_blank(),
    axis.text.x = element_text(color = "black", size = 10),
    axis.text.y = element_text(color = "black", size = 8),
    axis.title.y = element_text(vjust = 1, margin = margin(t = 0, r = 0, b = 0, l = 0, unit = "cm"), size = 10),
    plot.margin = unit(c(1, 5, 0, 0), units = "mm"),
    plot.title = element_text(size = 8)
  ) +
  guides(color = guide_legend(title.position = "top", title.hjust = 0.5))

```


```{r}
GO_endometrium_6_plot <- ggplot(data = GO_BP_Cats_endometrium_corr_pwbc_6[1:10, ],
                             aes(x = fold_enrichment,
                                y = term)) +
  geom_point() +
  coord_cartesian() +
  scale_colour_gradient(name = "Fold enrichment",
                        low = "#B8B8B8",
                        high = "#080808",
                        na.value = NA) +
  scale_x_continuous(name = "Fold enrichment") +
  scale_y_discrete(name = "Biological process") +
  theme_classic() +
  ggtitle("endometrium 6 months - Biological processes") +
  theme(
    legend.direction = "horizontal",
    legend.position = c(0.9, 1),
    legend.key.size = unit(0.5, "mm"),
    legend.text = element_text(size = 5, margin = margin(t = 0, r = 0, b = 0, l = 0, unit = "cm")),
    legend.title = element_text(size = 5),
    legend.background = element_blank(),
    axis.text.x = element_text(color = "black", size = 10),
    axis.text.y = element_text(color = "black", size = 8),
    axis.title.y = element_text(vjust = 1, margin = margin(t = 0, r = 0, b = 0, l = 0, unit = "cm"), size = 10),
    plot.margin = unit(c(1, 5, 0, 0), units = "mm"),
    plot.title = element_text(size = 8)
  ) +
  guides(color = guide_legend(title.position = "top", title.hjust = 0.5))
```


```{r}
pwbc_endometrium_6_summary_plot<-cowplot::plot_grid(
  cowplot::plot_grid(cowplot::plot_grid(cor_pwbc_endometrium_6_histogram, gene_connectivity_endometrium_6, gene_connectivity_pwbc_6, nrow =3, ncol=1, rel_heights=c(0.7,1,1),labels = c("A", "C","D"), label_fontface = "plain", label_size = 12 ),cowplot::plot_grid(plot_regr_6 , label_fontface = "plain", label_size = 12 , labels = c("B")) ,ncol=2,rel_widths=c(0.5,1)),
  cowplot::plot_grid(cowplot::plot_grid(GO_endometrium_6_plot,GO_pwbc_6_plot, nrow =2, align = 'v',labels = c("E", "F"), label_fontface = "plain", label_size = 12), NULL, rel_widths=c(1,0.5)),
  nrow=2, rel_heights=c(1,0.6))
```

```{r}
pwbc_endometrium_6_summary_plot
```


```{r}
ggsave(filename="/home/yassine/Downloads/between_tissue/Figures/pwbc_endometrium_6_summary_plot.png",
       plot = pwbc_endometrium_6_summary_plot, 
       width = 10, 
       height = 10)
```



# Test endometrium genes (significantly co-expressed with pwbc) for enrichment of biological processes.

```{r}
all_genes_endometrium_9 <-data.frame( gene=rownames(endometrium_9_count_filtered),
                                    stringsAsFactors=FALSE )

rownames(all_genes_endometrium_9)<-all_genes_endometrium_9$gene
N_expressed_genes<-length(all_genes_endometrium_9$gene)

gene.length_endometrium_9 <-gene.length[gene.length$ensembl_gene_id %in% all_genes_endometrium_9$gene,]

annotation.genelength.biomart_vector_endometrium_9 <-gene.length_endometrium_9$transcript_length
names(annotation.genelength.biomart_vector_endometrium_9)<-gene.length_endometrium_9$ensembl_gene_id

annotation.GO.BP.biomart<-annotation.GO.biomart[annotation.GO.biomart$namespace_1003=="biological_process", c(1,3)]
annotation.GO.BP.biomart<-annotation.GO.BP.biomart[annotation.GO.BP.biomart$ensembl_gene_id %in% rownames(all_genes_endometrium_9),]
annotation.GO.MF.biomart<-annotation.GO.biomart[annotation.GO.biomart$namespace_1003=="molecular_function", c(1,3)]
annotation.GO.MF.biomart<-annotation.GO.MF.biomart[annotation.GO.MF.biomart$ensembl_gene_id %in% rownames(all_genes_endometrium_9),]

```

```{r}
test.genes_endometrium_9 <-data.frame(
  a=unique(cor_pwbc_endometrium_9_[abs(cor_pwbc_endometrium_9_$correlation)>0.9]$gene_endometrium), 
  stringsAsFactors=FALSE)

all_genes_endometrium_9_numeric<-as.integer(all_genes_endometrium_9$gene %in% test.genes_endometrium_9$a)
names(all_genes_endometrium_9_numeric)<-all_genes_endometrium_9$gene
N_sig_genes<-length(test.genes_endometrium_9$a)

```

```{r}
set.seed(88972)

pwf_endometrium_9 <- nullp(all_genes_endometrium_9_numeric,
                            bias.data=annotation.genelength.biomart_vector_endometrium_9,
                            plot.fit=FALSE )
```


```{r}
GO_BP_Cats_endometrium_corr_pwbc_9 <- goseq(pwf_endometrium_9,
                                                  gene2cat=annotation.GO.BP.biomart,
                                                  method ="Sampling",
                                                  repcnt = 5000,
                                                  use_genes_without_cat=FALSE)
```

```{r}
write.table(GO_BP_Cats_endometrium_corr_pwbc_9,
            file = "/home/yassine/Downloads/between_tissue/GO_BP_Cats_endometrium_corr_pwbc_9.txt",
            quote = TRUE,
            sep = "\t",
            row.names = TRUE,
            col.names = TRUE) 
```

```{r}

GO_BP_Cats_endometrium_corr_pwbc_9<-GO_BP_Cats_endometrium_corr_pwbc_9[GO_BP_Cats_endometrium_corr_pwbc_9$numInCat>3,] 

GO_BP_Cats_endometrium_corr_pwbc_9$FWER<-p.adjust(
  GO_BP_Cats_endometrium_corr_pwbc_9$over_represented_pvalue,
  method ="holm")

GO_BP_Cats_endometrium_corr_pwbc_9<-GO_BP_Cats_endometrium_corr_pwbc_9[with(GO_BP_Cats_endometrium_corr_pwbc_9,order(FWER,over_represented_pvalue,-numInCat)), ]

GO_BP_Cats_endometrium_corr_pwbc_9$fold_enrichment<-(GO_BP_Cats_endometrium_corr_pwbc_9$numDEInCat/N_sig_genes)/(GO_BP_Cats_endometrium_corr_pwbc_9$numInCat/N_expressed_genes)
```



```{r}
annotation.GO.BP.biomart_testgenes<-annotation.GO.BP.biomart[annotation.GO.BP.biomart$ensembl_gene_id %in% test.genes_endometrium_9$a, ]

GO_BP_Cats_endometrium_corr_pwbc_9<-merge(GO_BP_Cats_endometrium_corr_pwbc_9,
                                                annotation.GO.BP.biomart_testgenes,
                                                by.x="category",
                                                by.y="go_id",
                                                all.x=FALSE,
                                                all.y=TRUE)

GO_BP_Cats_endometrium_corr_pwbc_9<-merge(GO_BP_Cats_endometrium_corr_pwbc_9,
                                                annotation.ensembl.symbol,
                                                by.x="ensembl_gene_id",
                                                by.y="ensembl_gene_id", 
                                                all=FALSE, 
                                                all.x=TRUE,
                                                all.y=FALSE)
```


```{r}
GO_BP_Cats_endometrium_corr_pwbc_9 <- arrange(GO_BP_Cats_endometrium_corr_pwbc_9, desc(fold_enrichment))
```


```{r}

GO_BP_Cats_endometrium_corr_pwbc_9$term<-factor(
  GO_BP_Cats_endometrium_corr_pwbc_9$term,
  levels=c(rev(unique(GO_BP_Cats_endometrium_corr_pwbc_9$term))))
```


# Test pwbc genes (significantly co-expressed with endometrium cells) for enrichment of biological processes.

```{r}
all_genes_pwbc_9 <-data.frame( gene=rownames(pwbc_9_count_filtered),
                                    stringsAsFactors=FALSE )

rownames(all_genes_pwbc_9) <- all_genes_pwbc_9$gene
N_expressed_genes <- length(all_genes_pwbc_9$gene)

gene.length_pwbc_9 <- gene.length[gene.length$ensembl_gene_id %in% all_genes_pwbc_9$gene,]

annotation.genelength.biomart_vector_pwbc_9 <- gene.length_pwbc_9$transcript_length
names(annotation.genelength.biomart_vector_pwbc_9) <- gene.length_pwbc_9$ensembl_gene_id

annotation.GO.BP.biomart<-annotation.GO.biomart[annotation.GO.biomart$namespace_1003=="biological_process", c(1,3)]

annotation.GO.BP.biomart<-annotation.GO.BP.biomart[annotation.GO.BP.biomart$ensembl_gene_id %in% rownames(all_genes_pwbc_9),]

annotation.GO.MF.biomart<-annotation.GO.biomart[annotation.GO.biomart$namespace_1003=="molecular_function", c(1,3)]

annotation.GO.MF.biomart<-annotation.GO.MF.biomart[annotation.GO.MF.biomart$ensembl_gene_id %in% rownames(all_genes_pwbc_9),]
```


```{r}
test.genes_pwbc_9 <-data.frame(
  a=unique(cor_pwbc_endometrium_9_[abs(cor_pwbc_endometrium_9_$correlation)>0.9]$gene_pwbc),
  stringsAsFactors=FALSE)

all_genes_pwbc_9_numeric<-as.integer(all_genes_pwbc_9$gene %in% test.genes_pwbc_9$a)
names(all_genes_pwbc_9_numeric)<-all_genes_pwbc_9$gene

N_sig_genes<-length(test.genes_pwbc_9$a)
```



```{r}
set.seed(87175)

pwf_pwbc_9 <-nullp(all_genes_pwbc_9_numeric,
                        bias.data=annotation.genelength.biomart_vector_pwbc_9,
                        plot.fit=FALSE ) 

GO_BP_Cats_pwbc_corr_endometrium_9 <-goseq(pwf_pwbc_9,
                                                 gene2cat=annotation.GO.BP.biomart,
                                                 method ="Sampling",
                                                 repcnt = 5000,
                                                 use_genes_without_cat=FALSE)
```


```{r}
GO_BP_Cats_pwbc_corr_endometrium_9<-GO_BP_Cats_pwbc_corr_endometrium_9[GO_BP_Cats_pwbc_corr_endometrium_9$numInCat>3,]

GO_BP_Cats_pwbc_corr_endometrium_9$FWER<-p.adjust(
  GO_BP_Cats_pwbc_corr_endometrium_9$over_represented_pvalue, 
  method ="holm")

GO_BP_Cats_pwbc_corr_endometrium_9<-GO_BP_Cats_pwbc_corr_endometrium_9[
  with(GO_BP_Cats_pwbc_corr_endometrium_9,
       order(FWER,over_represented_pvalue,-numInCat)), ]

```



```{r}
GO_BP_Cats_pwbc_corr_endometrium_9$fold_enrichment<-(GO_BP_Cats_pwbc_corr_endometrium_9$numDEInCat/N_sig_genes)/(GO_BP_Cats_pwbc_corr_endometrium_9$numInCat/N_expressed_genes)
                                                                                                
annotation.GO.BP.biomart_testgenes<-annotation.GO.BP.biomart[annotation.GO.BP.biomart$ensembl_gene_id %in% test.genes_pwbc_9$a, ]
```


```{r}
GO_BP_Cats_pwbc_corr_endometrium_9<-merge(GO_BP_Cats_pwbc_corr_endometrium_9,
                                              annotation.GO.BP.biomart_testgenes,
                                              by.x="category",
                                              by.y="go_id",
                                              all.x=TRUE,
                                              all.y=FALSE)

GO_BP_Cats_pwbc_corr_endometrium_9<-merge(GO_BP_Cats_pwbc_corr_endometrium_9,
                                              annotation.ensembl.symbol,
                                              by.x="ensembl_gene_id",
                                              by.y="ensembl_gene_id",
                                              all=FALSE, 
                                              all.x=TRUE,
                                              all.y=FALSE)
```




```{r}
GO_BP_Cats_pwbc_corr_endometrium_9 <- arrange(GO_BP_Cats_pwbc_corr_endometrium_9, desc(fold_enrichment))
```

```{r}
GO_BP_Cats_pwbc_corr_endometrium_9$term<-factor(
  GO_BP_Cats_pwbc_corr_endometrium_9$term,
  levels=c(rev(unique(GO_BP_Cats_pwbc_corr_endometrium_9$term))))
```


```{r}
GO_pwbc_9_plot <- ggplot(data = GO_BP_Cats_pwbc_corr_endometrium_9[1:10, ],
                             aes(x = fold_enrichment,
                                y = term)) +
  geom_point() +
  coord_cartesian() +
  scale_colour_gradient(name = "Fold enrichment",
                        low = "#B8B8B8",
                        high = "#080808",
                        na.value = NA) +
  scale_x_continuous(name = "Fold enrichment") +
  scale_y_discrete(name = "Biological process") +
  theme_classic() +
  ggtitle("pwbc 9 months - Biological processes") +
  theme(
    legend.direction = "horizontal",
    legend.position = c(0.9, 1),
    legend.key.size = unit(0.5, "mm"),
    legend.text = element_text(size = 5, margin = margin(t = 0, r = 0, b = 0, l = 0, unit = "cm")),
    legend.title = element_text(size = 5),
    legend.background = element_blank(),
    axis.text.x = element_text(color = "black", size = 10),
    axis.text.y = element_text(color = "black", size = 8),
    axis.title.y = element_text(vjust = 1, margin = margin(t = 0, r = 0, b = 0, l = 0, unit = "cm"), size = 10),
    plot.margin = unit(c(1, 5, 0, 0), units = "mm"),
    plot.title = element_text(size = 8)
  ) +
  guides(color = guide_legend(title.position = "top", title.hjust = 0.5))

```


```{r}
GO_endometrium_9_plot <- ggplot(data = GO_BP_Cats_endometrium_corr_pwbc_9[1:10, ],
                             aes(x = fold_enrichment,
                                y = term)) +
  geom_point() +
  coord_cartesian() +
  scale_colour_gradient(name = "Fold enrichment",
                        low = "#B8B8B8",
                        high = "#080808",
                        na.value = NA) +
  scale_x_continuous(name = "Fold enrichment") +
  scale_y_discrete(name = "Biological process") +
  theme_classic() +
  ggtitle("endometrium 9 months - Biological processes") +
  theme(
    legend.direction = "horizontal",
    legend.position = c(0.9, 1),
    legend.key.size = unit(0.5, "mm"),
    legend.text = element_text(size = 5, margin = margin(t = 0, r = 0, b = 0, l = 0, unit = "cm")),
    legend.title = element_text(size = 5),
    legend.background = element_blank(),
    axis.text.x = element_text(color = "black", size = 10),
    axis.text.y = element_text(color = "black", size = 8),
    axis.title.y = element_text(vjust = 1, margin = margin(t = 0, r = 0, b = 0, l = 0, unit = "cm"), size = 10),
    plot.margin = unit(c(1, 5, 0, 0), units = "mm"),
    plot.title = element_text(size = 8)
  ) +
  guides(color = guide_legend(title.position = "top", title.hjust = 0.5))
```

```{r}
pwbc_endometrium_9_summary_plot<-cowplot::plot_grid(
  cowplot::plot_grid(cowplot::plot_grid(cor_pwbc_endometrium_9_histogram, gene_connectivity_endometrium_9, gene_connectivity_pwbc_9, nrow =3, ncol=1, rel_heights=c(0.7,1,1),labels = c("A","C","D"), label_fontface = "plain", label_size = 12 ),cowplot::plot_grid(plot_regr_9 , label_fontface = "plain", label_size = 12 , labels = c("B")) ,ncol=2,rel_widths=c(0.5,1)),
  cowplot::plot_grid(cowplot::plot_grid(GO_endometrium_9_plot,GO_pwbc_9_plot, nrow =2, align = 'v',labels = c("E", "F"), label_fontface = "plain", label_size = 12), NULL, rel_widths=c(1,0.5)),
  nrow=2, rel_heights=c(1,0.6))
```

```{r}
pwbc_endometrium_9_summary_plot
```

```{r}
ggsave(filename="/home/yassine/Downloads/between_tissue/Figures/pwbc_endometrium_9_summary_plot.png",
       plot = pwbc_endometrium_9_summary_plot, 
       width = 10, 
       height = 10)
```


# Test endometrium genes (significantly co-expressed with pwbc) for enrichment of biological processes.

```{r}
all_genes_endometrium_19 <-data.frame( gene=rownames(endometrium_19_count_filtered),
                                    stringsAsFactors=FALSE )

rownames(all_genes_endometrium_19)<-all_genes_endometrium_19$gene
N_expressed_genes<-length(all_genes_endometrium_19$gene)

gene.length_endometrium_19 <-gene.length[gene.length$ensembl_gene_id %in% all_genes_endometrium_19$gene,]

annotation.genelength.biomart_vector_endometrium_19 <-gene.length_endometrium_19$transcript_length
names(annotation.genelength.biomart_vector_endometrium_19)<-gene.length_endometrium_19$ensembl_gene_id

annotation.GO.BP.biomart<-annotation.GO.biomart[annotation.GO.biomart$namespace_1003=="biological_process", c(1,3)]
annotation.GO.BP.biomart<-annotation.GO.BP.biomart[annotation.GO.BP.biomart$ensembl_gene_id %in% rownames(all_genes_endometrium_19),]
annotation.GO.MF.biomart<-annotation.GO.biomart[annotation.GO.biomart$namespace_1003=="molecular_function", c(1,3)]
annotation.GO.MF.biomart<-annotation.GO.MF.biomart[annotation.GO.MF.biomart$ensembl_gene_id %in% rownames(all_genes_endometrium_19),]

```

```{r}
test.genes_endometrium_19 <-data.frame(
  a=unique(cor_pwbc_endometrium_19_[abs(cor_pwbc_endometrium_19_$correlation)>0.9]$gene_endometrium), 
  stringsAsFactors=FALSE)

all_genes_endometrium_19_numeric<-as.integer(all_genes_endometrium_19$gene %in% test.genes_endometrium_19$a)
names(all_genes_endometrium_19_numeric)<-all_genes_endometrium_19$gene
N_sig_genes<-length(test.genes_endometrium_19$a)

```

```{r}
set.seed(88972)

pwf_endometrium_19 <- nullp(all_genes_endometrium_19_numeric,
                            bias.data=annotation.genelength.biomart_vector_endometrium_19,
                            plot.fit=FALSE )
```


```{r}
GO_BP_Cats_endometrium_corr_pwbc_19 <- goseq(pwf_endometrium_19,
                                                  gene2cat=annotation.GO.BP.biomart,
                                                  method ="Sampling",
                                                  repcnt = 5000,
                                                  use_genes_without_cat=FALSE)
```

```{r}
write.table(GO_BP_Cats_endometrium_corr_pwbc_19,
            file = "/home/yassine/Downloads/between_tissue/GO_BP_Cats_endometrium_corr_pwbc_19.txt",
            quote = TRUE,
            sep = "\t",
            row.names = TRUE,
            col.names = TRUE) 
```

```{r}

GO_BP_Cats_endometrium_corr_pwbc_19<-GO_BP_Cats_endometrium_corr_pwbc_19[GO_BP_Cats_endometrium_corr_pwbc_19$numInCat>3,] 

GO_BP_Cats_endometrium_corr_pwbc_19$FWER<-p.adjust(
  GO_BP_Cats_endometrium_corr_pwbc_19$over_represented_pvalue,
  method ="holm")

GO_BP_Cats_endometrium_corr_pwbc_19<-GO_BP_Cats_endometrium_corr_pwbc_19[with(GO_BP_Cats_endometrium_corr_pwbc_19,order(FWER,over_represented_pvalue,-numInCat)), ]

GO_BP_Cats_endometrium_corr_pwbc_19$fold_enrichment<-(GO_BP_Cats_endometrium_corr_pwbc_19$numDEInCat/N_sig_genes)/(GO_BP_Cats_endometrium_corr_pwbc_19$numInCat/N_expressed_genes)
```



```{r}
annotation.GO.BP.biomart_testgenes<-annotation.GO.BP.biomart[annotation.GO.BP.biomart$ensembl_gene_id %in% test.genes_endometrium_19$a, ]

GO_BP_Cats_endometrium_corr_pwbc_19<-merge(GO_BP_Cats_endometrium_corr_pwbc_19,
                                                annotation.GO.BP.biomart_testgenes,
                                                by.x="category",
                                                by.y="go_id",
                                                all.x=FALSE,
                                                all.y=TRUE)

GO_BP_Cats_endometrium_corr_pwbc_19<-merge(GO_BP_Cats_endometrium_corr_pwbc_19,
                                                annotation.ensembl.symbol,
                                                by.x="ensembl_gene_id",
                                                by.y="ensembl_gene_id", 
                                                all=FALSE, 
                                                all.x=TRUE,
                                                all.y=FALSE)
```

```{r}

GO_BP_Cats_endometrium_corr_pwbc_19 <- arrange(GO_BP_Cats_endometrium_corr_pwbc_19, desc(fold_enrichment))
```


```{r}

GO_BP_Cats_endometrium_corr_pwbc_19$term<-factor(
  GO_BP_Cats_endometrium_corr_pwbc_19$term,
  levels=c(rev(unique(GO_BP_Cats_endometrium_corr_pwbc_19$term))))
```


# Test pwbc genes (significantly co-expressed with endometrium cells) for enrichment of biological processes.

```{r}
all_genes_pwbc_19 <-data.frame( gene=rownames(pwbc_19_count_filtered),
                                    stringsAsFactors=FALSE )

rownames(all_genes_pwbc_19) <- all_genes_pwbc_19$gene
N_expressed_genes <- length(all_genes_pwbc_19$gene)

gene.length_pwbc_19 <- gene.length[gene.length$ensembl_gene_id %in% all_genes_pwbc_19$gene,]

annotation.genelength.biomart_vector_pwbc_19 <- gene.length_pwbc_19$transcript_length
names(annotation.genelength.biomart_vector_pwbc_19) <- gene.length_pwbc_19$ensembl_gene_id

annotation.GO.BP.biomart<-annotation.GO.biomart[annotation.GO.biomart$namespace_1003=="biological_process", c(1,3)]

annotation.GO.BP.biomart<-annotation.GO.BP.biomart[annotation.GO.BP.biomart$ensembl_gene_id %in% rownames(all_genes_pwbc_19),]

annotation.GO.MF.biomart<-annotation.GO.biomart[annotation.GO.biomart$namespace_1003=="molecular_function", c(1,3)]

annotation.GO.MF.biomart<-annotation.GO.MF.biomart[annotation.GO.MF.biomart$ensembl_gene_id %in% rownames(all_genes_pwbc_19),]
```


```{r}
test.genes_pwbc_19 <-data.frame(
  a=unique(cor_pwbc_endometrium_19_[abs(cor_pwbc_endometrium_19_$correlation)>0.9]$gene_pwbc),
  stringsAsFactors=FALSE)

all_genes_pwbc_19_numeric<-as.integer(all_genes_pwbc_19$gene %in% test.genes_pwbc_19$a)
names(all_genes_pwbc_19_numeric)<-all_genes_pwbc_19$gene

N_sig_genes<-length(test.genes_pwbc_19$a)
```



```{r}
set.seed(87175)

pwf_pwbc_19 <-nullp(all_genes_pwbc_19_numeric,
                        bias.data=annotation.genelength.biomart_vector_pwbc_19,
                        plot.fit=FALSE ) 

GO_BP_Cats_pwbc_corr_endometrium_19 <-goseq(pwf_pwbc_19,
                                                 gene2cat=annotation.GO.BP.biomart,
                                                 method ="Sampling",
                                                 repcnt = 5000,
                                                 use_genes_without_cat=FALSE)
```


```{r}
GO_BP_Cats_pwbc_corr_endometrium_19<-GO_BP_Cats_pwbc_corr_endometrium_19[GO_BP_Cats_pwbc_corr_endometrium_19$numInCat>3,]

GO_BP_Cats_pwbc_corr_endometrium_19$FWER<-p.adjust(
  GO_BP_Cats_pwbc_corr_endometrium_19$over_represented_pvalue, 
  method ="holm")

GO_BP_Cats_pwbc_corr_endometrium_19<-GO_BP_Cats_pwbc_corr_endometrium_19[
  with(GO_BP_Cats_pwbc_corr_endometrium_19,
       order(FWER,over_represented_pvalue,-numInCat)), ]

```



```{r}
GO_BP_Cats_pwbc_corr_endometrium_19$fold_enrichment<-(GO_BP_Cats_pwbc_corr_endometrium_19$numDEInCat/N_sig_genes)/(GO_BP_Cats_pwbc_corr_endometrium_19$numInCat/N_expressed_genes)
                                                                                                
annotation.GO.BP.biomart_testgenes<-annotation.GO.BP.biomart[annotation.GO.BP.biomart$ensembl_gene_id %in% test.genes_pwbc_19$a, ]
```


```{r}
GO_BP_Cats_pwbc_corr_endometrium_19<-merge(GO_BP_Cats_pwbc_corr_endometrium_19,
                                              annotation.GO.BP.biomart_testgenes,
                                              by.x="category",
                                              by.y="go_id",
                                              all.x=TRUE,
                                              all.y=FALSE)

GO_BP_Cats_pwbc_corr_endometrium_19<-merge(GO_BP_Cats_pwbc_corr_endometrium_19,
                                              annotation.ensembl.symbol,
                                              by.x="ensembl_gene_id",
                                              by.y="ensembl_gene_id",
                                              all=FALSE, 
                                              all.x=TRUE,
                                              all.y=FALSE)
```


```{r}
GO_BP_Cats_pwbc_corr_endometrium_19 <- arrange(GO_BP_Cats_pwbc_corr_endometrium_19, desc(fold_enrichment))
```


```{r}
GO_BP_Cats_pwbc_corr_endometrium_19$term<-factor(
  GO_BP_Cats_pwbc_corr_endometrium_19$term,
  levels=c(rev(unique(GO_BP_Cats_pwbc_corr_endometrium_19$term))))

```


```{r}
GO_pwbc_19_plot <- ggplot(data = GO_BP_Cats_pwbc_corr_endometrium_19[1:10, ],
                             aes(x = fold_enrichment,
                                y = term)) +
  geom_point() +
  coord_cartesian() +
  scale_colour_gradient(name = "Fold enrichment",
                        low = "#B8B8B8",
                        high = "#080808",
                        na.value = NA) +
  scale_x_continuous(name = "Fold enrichment") +
  scale_y_discrete(name = "Biological process") +
  theme_classic() +
  ggtitle("pwbc 19 months - Biological processes") +
  theme(
    legend.direction = "horizontal",
    legend.position = c(0.9, 1),
    legend.key.size = unit(0.5, "mm"),
    legend.text = element_text(size = 5, margin = margin(t = 0, r = 0, b = 0, l = 0, unit = "cm")),
    legend.title = element_text(size = 5),
    legend.background = element_blank(),
    axis.text.x = element_text(color = "black", size = 10),
    axis.text.y = element_text(color = "black", size = 8),
    axis.title.y = element_text(vjust = 1, margin = margin(t = 0, r = 0, b = 0, l = 0, unit = "cm"), size = 10),
    plot.margin = unit(c(1, 5, 0, 0), units = "mm"),
    plot.title = element_text(size = 8)
  ) +
  guides(color = guide_legend(title.position = "top", title.hjust = 0.5))

```


```{r}
GO_endometrium_19_plot <- ggplot(data = GO_BP_Cats_endometrium_corr_pwbc_19[1:10, ],
                             aes(x = fold_enrichment,
                                y = term)) +
  geom_point() +
  coord_cartesian() +
  scale_colour_gradient(name = "Fold enrichment",
                        low = "#B8B8B8",
                        high = "#080808",
                        na.value = NA) +
  scale_x_continuous(name = "Fold enrichment") +
  scale_y_discrete(name = "Biological process") +
  theme_classic() +
  ggtitle("endometrium 19 months - Biological processes") +
  theme(
    legend.direction = "horizontal",
    legend.position = c(0.9, 1),
    legend.key.size = unit(0.5, "mm"),
    legend.text = element_text(size = 5, margin = margin(t = 0, r = 0, b = 0, l = 0, unit = "cm")),
    legend.title = element_text(size = 5),
    legend.background = element_blank(),
    axis.text.x = element_text(color = "black", size = 10),
    axis.text.y = element_text(color = "black", size = 8),
    axis.title.y = element_text(vjust = 1, margin = margin(t = 0, r = 0, b = 0, l = 0, unit = "cm"), size = 10),
    plot.margin = unit(c(1, 5, 0, 0), units = "mm"),
    plot.title = element_text(size = 8)
  ) +
  guides(color = guide_legend(title.position = "top", title.hjust = 0.5))
```

```{r}
pwbc_endometrium_19_summary_plot<-cowplot::plot_grid(
  cowplot::plot_grid(cowplot::plot_grid(cor_pwbc_endometrium_19_histogram, gene_connectivity_endometrium_19, gene_connectivity_pwbc_19, nrow =3, ncol=1, rel_heights=c(0.7,1,1),labels = c("A", "C","D"), label_fontface = "plain", label_size = 12 ),cowplot::plot_grid(plot_regr_19 , label_fontface = "plain", label_size = 12 , labels = c("B")) ,ncol=2,rel_widths=c(0.5,1)),
  cowplot::plot_grid(cowplot::plot_grid(GO_endometrium_19_plot,GO_pwbc_19_plot, nrow =2, align = 'v',labels = c("E", "F"), label_fontface = "plain", label_size = 12), NULL, rel_widths=c(1,0.5)),
  nrow=2, rel_heights=c(1,0.6))
```


```{r}
pwbc_endometrium_19_summary_plot
```

```{r}
ggsave(filename="/home/yassine/Downloads/between_tissue/Figures/pwbc_endometrium_19_summary_plot.png",
       plot = pwbc_endometrium_19_summary_plot, 
       width = 10, 
       height = 10)
```
